code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -123,3 +123,104 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 	if(CHECK_BITFIELD(direction_flag, CONE_PART_MIDDLE_DIAG))
 		do_acid_cone_spray(next_normal_turf, distance_left - 1, facing, CONE_PART_DIAG_LEFT|CONE_PART_DIAG_RIGHT, spray)
 		do_acid_cone_spray(next_normal_turf, distance_left - 2, facing, (distance_left < 5) ? CONE_PART_MIDDLE : CONE_PART_MIDDLE_DIAG, spray)
+
+// ***************************************
+// *********** Acid dash
+// ***************************************
+/datum/action/xeno_action/activable/acid_dash
+	name = "Acid Dash"
+	action_icon_state = "charge"
+	mechanics_text = "Instantly dash, tackling the first marine in your path. If you manage to tackle someone, gain another weaker cast of the ability."
+	ability_name = "acid dash"
+	plasma_cost = 250
+	cooldown_timer = 30 SECONDS
+	//How far can we dash
+	var/range = 5
+	//Can we use the ability again
+	var/recast_available = FALSE
+	//Is this the recast
+	var/recast = FALSE
+	//The last tile we dashed through, used when swapping with a human
+	var/turf/last_turf
+
+/datum/action/xeno_action/activable/acid_dash/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+/datum/action/xeno_action/activable/acid_dash/on_cooldown_finish()
+	to_chat(owner, span_xenodanger("Our exoskeleton quivers as we get ready to use Acid Dash again."))
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 50, 0, 1)
+	return ..()
+
+/datum/action/xeno_action/activable/acid_dash/proc/dash_complete()
+	SIGNAL_HANDLER
+	if(recast_available && !recast)
+		addtimer(CALLBACK(src, /datum/action/xeno_action/proc/add_cooldown), 2 SECONDS) //Recast 'window', how long you have to press the ability button again
+		recast = TRUE
+	else
+		recast_available = FALSE
+		recast = FALSE
+		add_cooldown()
+	UnregisterSignal(owner, list(COMSIG_XENO_OBJ_THROW_HIT, COMSIG_XENO_NONE_THROW_HIT, COMSIG_XENO_LIVING_THROW_HIT, COMSIG_MOVABLE_MOVED))
+
+/datum/action/xeno_action/activable/acid_dash/proc/mob_hit(datum/source, mob/M)
+	SIGNAL_HANDLER
+	if(recast) //That's the recast, we don't stop for mobs
+		return COMPONENT_KEEP_THROWING
+	if(ishuman(M)) //It's the first cast, we swap with the first human in our path
+		var/mob/living/carbon/human/target = M
+		var/mob/living/carbon/xenomorph/X = owner
+		var/turf/victim_turf = get_turf(target)
+		var/turf/aggressor_turf = last_turf
+
+		target.throw_at(aggressor_turf, 1, 1, X)
```

Comment:
why a throw?

another point, why a shuffle? I honestly feel like this is just a nest for bugs, even though i can't put my finger on why. If you want a shuffle i'd much prefer you make your own "throw" proc, using normal move

Reply:
From the xeno-player standpoint a shuffle is much more interesting since it allows you to instantly bodyblock whoever you tackled. For example if you dash through a marine with harm intent on in a 1-tile wide corridor in a maze they'll already be blocked, allowing you to spit/slash them, the knockdown is also extremely short at 1/2 a second so I fear getting anything out of it would be hard with helping a bit.